client/my-sites/stats/stats-date-control/index.tsx

```diff
@@ -102,10 +102,51 @@ const StatsDateControl = ( { slug, queryParams }: StatsDateControlProps ) => {
 		page( generateNewLink( shortcut.period, startDate, endDate ) );
 	};
 
+	const getShortcutForRange = () => {
+		const today = moment().format( 'YYYY-MM-DD' );
+		// Today
+		if ( today === dateRange.chartEnd && dateRange.daysInRange === 1 ) {
+			return shortcutList[ 0 ];
+		}
+		// Last 7 days
+		if ( today === dateRange.chartEnd && dateRange.daysInRange === 7 ) {
+			return shortcutList[ 2 ];
+		}
+		// Last 30 days
+		if ( today === dateRange.chartEnd && dateRange.daysInRange === 30 ) {
+			return shortcutList[ 3 ];
+		}
+		// Last year
+		if ( today === dateRange.chartEnd && dateRange.daysInRange === 365 ) {
+			return shortcutList[ 4 ];
+		}
+		const yesterday = moment().subtract( 1, 'days' ).format( 'YYYY-MM-DD' );
+		// Yesterday
+		if ( yesterday === dateRange.chartEnd && dateRange.daysInRange === 1 ) {
+			return shortcutList[ 1 ];
+		}
+		return null;
+	};
+
+	const getButtonLabel = () => {
+		// Test for a shortcut match.
+		const shortcut = getShortcutForRange();
+		if ( shortcut !== null ) {
+			return shortcut.label;
+		}
+		// Generate a full date range for the label.
+		const startDate = moment( dateRange.chartStart ).format( 'MMMM Do, YYYY' );
```

Comment:
I think this format doesn't take localisation into account.

Reply:
If we couldn't localise it, storing the string as a constant will improve ways to find those instances.

Reply:
I'm not sure how to test this. After the localization changes you added it does produce a localized version of the date string but I'm not sure how "correct" it is for native speakers. I'll do some reading up on this and check back in.